scripts/e2e/compare_metrics.py

```diff
@@ -58,9 +93,12 @@ def parse_metrics(content):
     for family in text_string_to_metric_families(content):
         for sample in family.samples:
             labels = dict(sample.labels)
-            #simply pop undesirable metric labels
-            labels.pop('service_instance_id',None)
             
+            should_exclude= should_exclude_metric(sample.name, labels)
+            if should_exclude:
```

Comment:
```suggestion
            if should_exclude_metric(sample.name, labels):
```

Reply:
yes (although I am not sure excluding full metrics is a good thing, per the other comment).